src/middlewared/middlewared/plugins/sysdataset.py

```diff
@@ -295,7 +295,10 @@ async def setup(self, mount, exclude_pool=None):
             if osc.IS_LINUX:
                 await self.middleware.call('etc.generate', 'glusterd')
 
-            await self.middleware.call('smb.configure')
+            await self.middleware.call('smb.setup_directories')
+            # The following should be backgrounded since they may be quite
+            # long-running.
```

Comment:
A bit confused, why dont we do that, then?

Reply:
Ah, they are jobs already, nevermind

Reply:
Because it's a job and we don't wait for it to complete